modules/js-and-css/speculation-rules/class-plsr-url-pattern-prefixer.php

```diff
@@ -86,8 +86,32 @@ public function prefix_path_pattern( string $path_pattern, string $context = 'ho
 	 */
 	public static function get_default_contexts(): array {
 		return array(
-			'home' => trailingslashit( wp_parse_url( home_url( '/' ), PHP_URL_PATH ) ),
-			'site' => trailingslashit( wp_parse_url( site_url( '/' ), PHP_URL_PATH ) ),
+			'home' => self::escape_pattern_string( trailingslashit( wp_parse_url( home_url( '/' ), PHP_URL_PATH ) ) ),
+			'site' => self::escape_pattern_string( trailingslashit( wp_parse_url( site_url( '/' ), PHP_URL_PATH ) ) ),
```

Comment:
This escapes the "path prefixes", which makes sense. But wouldn't we also want to escape any other parts of the path string in the same way?

Any string that a 3P developer may provide via the `plsr_speculation_rules_href_exclude_paths` filter below could also contain one of those characters that would need to be escaped as far as I can tell.